lib/ansible/cli/playbook.py

```diff
@@ -76,19 +78,21 @@ def run(self):
 
         # initial error check, to make sure all specified playbooks are accessible
         # before we start running anything through the playbook executor
+
+        playbook_dirs = []
         for playbook in context.CLIARGS['args']:
             if not os.path.exists(playbook):
                 raise AnsibleError("the playbook: %s could not be found" % playbook)
             if not (os.path.isfile(playbook) or stat.S_ISFIFO(os.stat(playbook).st_mode)):
                 raise AnsibleError("the playbook: %s does not appear to be a file" % playbook)
+
+            playbook_dir = os.path.dirname(os.path.abspath(to_bytes(playbook, errors='surrogate_or_strict')))
```

Comment:
We don't have a policy/guideline around this yet, but anytime we set an intermediate variable explicitly to bytes, we should be prefixing with `b_` so that we understand the implications.

Reply:
It's also too late probably to worry about this now, but we should likely also be transitioning to only using bytes with the APIs that require them.  A full unicode sandwich would look like:

```
playbook_dir = os.path.dirname(to_text(os.path.abspath(to_bytes(playbook, errors='surrogate_or_strict')), errors='surrogate_or_strict'))
```

Our goal should be to not have long lived byte strings floating throughout the codebase.

Reply:
prefixed long-lived vars where it needs to be bytes or native